pkg/apis/application/v1alpha1/types.go

```diff
@@ -1816,6 +1816,17 @@ func (status *ApplicationStatus) GetErrorConditions() []ApplicationCondition {
 	return result
 }
 
+func (status *ApplicationStatus) GetConditions(conditionTypes map[ApplicationConditionType]bool) []ApplicationCondition {
```

Comment:
minor - this feels weird to me - why does a getter take a map? - why not an slice?
where is the test?

Reply:
This is because golang don't have sets :( I did not want to group give list into a map